### ISSUE TITLE
Fix deprecated assertFileNotExists() calls 

### ISSUE THREAD

**Issue by dmitriim** [catalyst/moodle-auth_saml2#583]
In Moodle 3.11:
```
5) generator_testcase::test_create_idp_entity_no_files
assertFileNotExists() is deprecated and will be removed in PHPUnit 10. Refactor your code to use assertFileDoesNotExist() instead.

/var/www/html/lib/phpunit/classes/advanced_testcase.php:80

6) auth_saml2_metadata_writer_testcase::test_write_non_dataroot_path
assertFileNotExists() is deprecated and will be removed in PHPUnit 10. Refactor your code to use assertFileDoesNotExist() instead.

/var/www/html/lib/phpunit/classes/basic_testcase.php:64
```

**Comment by kabalin** [comment]
Yeah, noticed that, it fails in 3.9 and 3.10 CI if changed to `assertFileDoesNotExist` now.

**Comment by dmitriim** [comment]
that may be time to branch out for 3.11+... @brendanheywood @danmarsden ?

**Comment by danmarsden** [comment]
@dmitriim yeah - I had the same thought so I've just created one - if you get a chance can you please update the readme with the detail on the new branch and drop the patch in for PHPunit too?

**Comment by kabalin** [comment]
So, any new feature will miss < 3.11 (unless someone keep them in sync or send two PRs) and it is all because of deprication notice that does not affect test result yet in any way?

**Comment by danmarsden** [comment]
No - we don't follow upstream 'master branch for features only' policy. We backport features to supported branches where possible. But yes it does mean more work.

**Comment by kabalin** [comment]
Yeah, I got that, I mean adding more work when it is not strictly necessary (it is just a deprecation notice in test, not some new feature that does not work in versions lower than 3.11)  😃

**Comment by danmarsden** [comment]
for some reason I thought we were getting failures in there on 4.0/master - not just warnings - but the failure in the master branch might be something different.... We could hold off branching until 4.0 I guess

**Comment by danmarsden** [comment]
yeah - lets hold off branching this for now - deprecated warnings in test I can live with - @dmitriim - is this causing failures in your pipelines? - or just warnings?

**Comment by danmarsden** [comment]
fixed in #659 with backwards compatible code.
